src/app/sevices/task.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Task } from '../models/task';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService{
+  jsonSubscription: Subscription = Subscription.EMPTY;
+  tasksSubscription: Subscription = Subscription.EMPTY;
+  private tasksSubject: BehaviorSubject<Task[]> = new BehaviorSubject<Task[]>([]);
+  tasksObservable$: Observable<Task[]> = this.tasksSubject.asObservable();
+  tasksComplete$: BehaviorSubject<number> = new BehaviorSubject<number>(0);
+
+  constructor(private http: HttpClient) {
+    this.tasksSubscription = this.tasksSubject.subscribe((tasks: Task[]) => {});
+    this.jsonSubscription = this.http.get('assets/dummy-tasks.json').subscribe((res: any) => {
```

Comment:
usually you don't subscribe in the services, so you'd have this.http.get(url);

then you subscribe to this data in components where you can unsubscribe later, do that please